### PR TITLE
Sync with upstream master

### PR DESCRIPTION
**Pull request by bhuvankitman**
I have solved the conflicts caused by our changes on the fork.

Please refer [this](https://github.com/xmartlabs/XLForm/compare/master...KitmanLabs:XLForm:master) and make sure our changes persist in this PR

https://github.com/xmartlabs/XLForm/compare/master...KitmanLabs:XLForm:master

Ticket - https://kitmanlabs.atlassian.net/browse/MOB-1864

### CODE REVIEW

**Review by kittwebhook — CHANGES_REQUESTED** [review]
Hi @bhuvankitman, can you please make sure to link an issue in [Github Projects](https://github.com/KitmanLabs/projects/issues/) or [Jira](https://kitmanlabs.atlassian.net/) on this PR?

**Comment by kittwebhook** [comment]
I wasn't able to update XLForm#9 - check if the bots team has write access to it?

**Comment by kittwebhook** [comment]
Hi @bhuvankitman, can you please make sure to link an issue in [Github Projects](https://github.com/KitmanLabs/projects/issues/) or [Jira](https://kitmanlabs.atlassian.net/) on this PR?

**Review by NoamEfergan — APPROVED** [review]
> saw it's a library, so i haven't reviewed anything

LGTM :+1:

**Review by akankshakitman — APPROVED** [review]

**Review by ChrisLawther-KitmanLabs — APPROVED** [review]

File: XLForm/XL/XLFormDateCell.m [hunk 1]
```diff
@@ -73,7 +75,8 @@ -(BOOL)becomeFirstResponder
             XLFormRowDescriptor * datePickerRowDescriptor = [XLFormRowDescriptor formRowDescriptorWithTag:nil rowType:XLFormRowDescriptorTypeDatePicker];
             XLFormDatePickerCell * datePickerCell = (XLFormDatePickerCell *)[datePickerRowDescriptor cellForFormController:self.formViewController];
             [self setModeToDatePicker:datePickerCell.datePicker];
-            if (self.rowDescriptor.value){
+            [self setStyleToDatePicker:datePickerCell.datePicker];
```
Comment (nkennedyKit) [inline XLForm/XL/XLFormDateCell.m:78]: Looks like we lose the default date logic here? This is what we have in the forked version

```obj-c
            [self setStyleToDatePicker:datePickerCell.datePicker];
            if (self.rowDescriptor.value){                
                [datePickerCell.datePicker setDate:self.rowDescriptor.value animated:[self.rowDescriptor.rowType isEqualToString:XLFormRowDescriptorTypeCountDownTimerInline]];
            } else if (self.defaultDate) {
                [datePickerCell.datePicker setDate:self.defaultDate];
            }
```

**Review by nkennedyKit — CHANGES_REQUESTED** [review]
There is just one section where i'm not sure if the default date logic followed through or not.

File: XLForm/XL/XLFormDateCell.m (hunk 1 above)
Comment (bhuvankitman) [reply XLForm/XL/XLFormDateCell.m:78]: It is still there @nkennedyKit if you open the file. 

Only date new style is added here 
`[self setStyleToDatePicker:datePickerCell.datePicker];`

Complete code looks like this 
```
[self setModeToDatePicker:datePickerCell.datePicker];
[self setStyleToDatePicker:datePickerCell.datePicker]; <----- New code
if (self.rowDescriptor.value){
    [datePickerCell.datePicker setDate:self.rowDescriptor.value animated:[self.rowDescriptor.rowType isEqualToString:XLFormRowDescriptorTypeCountDownTimerInline]];
} else if (self.defaultDate) {
    [datePickerCell.datePicker setDate:self.defaultDate];
}

```

**Pull request merged**